### ISSUE TITLE
MacOS can't resolve CSS paths with AOT

### ISSUE THREAD

**Issue by crutchcorn** [NativeScript/nativescript-dev-webpack#400]
I have a Nativescript app that I've been developing on Windows that shouldn't have any Android specific code enabled. Because of this, my company got a MacBook in order to build and publish the app for iOS. However, while the code compiles fine on Windows (with the @next branch), it does not do so on MacOS.  The following error shows up whether I use `tns run` or `tns build`:
```
ERROR in Error: Couldn't resolve resource login.component-common.css relative to /Users/corbincrutchley/git/linkd-nativescript/app/modules/login/login.component.ts
app/main.aot.ts(4,36): error TS2307: Cannot find module './app.module.ngfactory'.
```

I'm importing the css in the following way, and you can see that the files are there properly. Is there something I'm doing wrong?
<img width="1280" alt="screen shot 2018-01-09 at 8 52 07 am" src="https://user-images.githubusercontent.com/9100169/34732786-17f0461c-f51b-11e7-809f-f0caed8ffe9d.png">

**Comment by sis0k0** [comment]
@crutchcorn, note that you can use directly the '.scss' file now. Also, can you share a simple app with reproduction of the problem?

**Comment by crutchcorn** [comment]
Sorry it took so long, and I also apologize for uploading as a ZIP, but it would take a bit to upload everything from my (minified) project into Playground or something similar

**Comment by jeffswitzer** [comment]
Somewhat related, since the tns 3.4 upgrade non of my scss/css styles show up when building with --bundle. When I don't use --bundle the css shows up fine. This is on windows 10 / android. The build show's no errors when making the apk. Here's my webpack config - I have app.scss and app.css files in the main app folder that no longer seem to be picked up:

```
const { resolve, join  } = require("path");

const webpack = require("webpack");
const nsWebpack = require("nativescript-dev-webpack");
const nativescriptTarget = require("nativescript-dev-webpack/nativescript-target");
const CopyWebpackPlugin = require("copy-webpack-plugin");
const ExtractTextPlugin = require("extract-text-webpack-plugin");
const { BundleAnalyzerPlugin } = require("webpack-bundle-analyzer");
const { NativeScriptWorkerPlugin } = require("nativescript-worker-loader/NativeScriptWorkerPlugin");

module.exports = env => {
    const platform = env && (env.android && "android" || env.ios && "ios");
    if (!platform) {
        throw new Error("You need to provide a target platform!");
    }
    const platforms = ["ios", "android"];
    const { snapshot, uglify, report, aot } = env;
    const ngToolsWebpackOptions = { tsConfigPath: "tsconfig.json" };

    const config = {
        context: resolve("./app"),
        target: nativescriptTarget,
        entry: {
            bundle: aot ? "./main.aot.ts" : "./main.ts",
            vendor: "./vendor",
        },
        output: {
            pathinfo: true,
            // Default destination inside platforms/<platform>/...
            path: resolve(nsWebpack.getAppPath(platform)),
            libraryTarget: "commonjs2",
            filename: "[name].js",
        },
        resolve: {
            extensions: [".ts", ".js", ".scss", ".css"],
            // Resolve {N} system modules from tns-core-modules
            modules: [
                "node_modules/tns-core-modules",
                "node_modules",
            ],
            alias: {
                '~': resolve("./app")
            },
            // don't resolve symlinks to symlinked modules
            symlinks: false
        },
        resolveLoader: {
            // don't resolve symlinks to symlinked loaders
            symlinks: false
        },
        node: {
            // Disable node shims that conflict with NativeScript
            "http": false,
            "timers": false,
            "setImmediate": false,
            "fs": "empty",
        },
        module: {
            rules: [
                { test: /\.html$|\.xml$/, use: "raw-loader" },

                // tns-core-modules reads the app.css and its imports using css-loader
                { test: /\/app\.css$/, use: "css-loader?url=false" },
                { test: /\/app\.scss$/, use: ["css-loader?url=false", "sass-loader"] },

                // Angular components reference css files and their imports using raw-loader
                { test: /\.css$/, exclude: /\/app\.css$/, use: "raw-loader" },
                { test: /\.scss$/, exclude: /\/app\.scss$/, use: ["raw-loader", "resolve-url-loader", "sass-loader"] },

                // Compile TypeScript files with ahead-of-time compiler.
                { test: /.ts$/, use: [
                    "nativescript-dev-webpack/moduleid-compat-loader",
                    { loader: "@ngtools/webpack", options: ngToolsWebpackOptions },
                ]},
            ],
        },
        plugins: [
            // Vendor libs go to the vendor.js chunk
            new webpack.optimize.CommonsChunkPlugin({
                name: ["vendor"],
            }),
            // Define useful constants like TNS_WEBPACK
            new webpack.DefinePlugin({
                "global.TNS_WEBPACK": "true",
            }),
            // Copy assets to out dir. Add your own globs as needed.
            new CopyWebpackPlugin([
                { from: "App_Resources/**" },
                { from: "fonts/**" },
                { from: "**/*.jpg" },
                { from: "**/*.png" },
                { from: "**/*.xml" },
            ]),
            // Generate a bundle starter script and activate it in package.json
            new nsWebpack.GenerateBundleStarterPlugin([
                "./vendor",
                "./bundle",
            ]),
            // Support for web workers since v3.2
            new NativeScriptWorkerPlugin(),
            // AngularCompilerPlugin with augmented NativeScript filesystem to handle platform specific resource resolution.
            new nsWebpack.NativeScriptAngularCompilerPlugin(
                Object.assign({
                    entryModule: resolve(__dirname, "app/app.module#AppModule"),
                    skipCodeGeneration: !aot,
                    platformOptions: {
                        platform,
                        platforms,
                        // ignore: ["App_Resources"]
                    },
                }, ngToolsWebpackOptions)
            ),
            // Does IPC communication with the {N} CLI to notify events when running in watch mode.
            new nsWebpack.WatchStateLoggerPlugin(),
        ],
    };
    if (report) {
        // Generate report files for bundles content
        config.plugins.push(new BundleAnalyzerPlugin({
            analyzerMode: "static",
            openAnalyzer: false,
            generateStatsFile: true,
            reportFilename: join(__dirname, "report", `report.html`),
            statsFilename: join(__dirname, "report", `stats.json`),
        }));
    }
    if (snapshot) {
        config.plugins.push(new nsWebpack.NativeScriptSnapshotPlugin({
            chunk: "vendor",
            projectRoot: __dirname,
            webpackConfig: config,
            targetArchs: ["arm", "arm64", "ia32"],
            tnsJavaClassesOptions: { packages: ["tns-core-modules" ] },
            useLibs: false
        }));
    }
    if (uglify) {
        config.plugins.push(new webpack.LoaderOptionsPlugin({ minimize: true }));

        // Work around an Android issue by setting compress = false
        const compress = platform !== "android";
        config.plugins.push(new webpack.optimize.UglifyJsPlugin({
            mangle: { except: nsWebpack.uglifyMangleExcludes },
            compress,
        }));
    }
    return config;
};
```

**Comment by crutchcorn** [comment]
I just double checked, and with the example I mentioned (assuming you add back some icons; sorry about removing those 😅 ) and it's both bundling fine and `tns run`ing fine. This is driving me absolutely bonkers. Let me try AGAIN to reproduce

**Comment by crutchcorn** [comment]
I just double checked, and with the example I mentioned (assuming you add back some icons; sorry about removing those 😅 ) and it's both bundling fine and `tns run`ing fine but breaks on AOT.

**Comment by sis0k0** [comment]
@jeffswitzer, the problem on windows should be fixed if you use the 'next' tag and will be released in the next version.

**Comment by sis0k0** [comment]
@crutchcorn, you need to add these fixes to your webpack configuration: https://github.com/NativeScript/nativescript-dev-webpack/pull/385/files#diff-12bf665255924268b4cf58020290b563L63.
Alternatively, you can run a script that will overwrite your webpack configuration with the one from the plugin:
```
./node_modules/.bin/update-ns-webpack --configs
```
Then, you'll have to apply your modifications (the workaround for ngrx) to the newly generated webpack.config.js

**Comment by crutchcorn** [comment]
Thank you for your support; sorry for filing such a silly bug report 😅 I'll make sure to look into that tool next time first

**Comment by jeffswitzer** [comment]
@sis0k0 thanks for confirming the issue.... after moving to next however I still experience the problem.

I've tried 

npm uninstall -g nativescript
npm install -g nativescript@next
"tns-android": { "version": "next"}
"nativescript-dev-webpack": "next",
tns platform remove android
tns platform add android@next

then 
tns build android --bundle --release --key-store-path mykey.jks --key-store-password mypass --key-store-alias myalias --key-store-alias-password mypass

After copying the apk to my phone, I still don't see the app.scss applied globally. If I add styleUrls['./app.scss] to individual components it picks up the style, but I'd prefer not to do that everywhere. Is there a closed bug for this somewhere that I should update?

**Comment by sis0k0** [comment]
@jeffswitzer, you'll have to apply the above mentioned fix, too - https://github.com/NativeScript/nativescript-dev-webpack/issues/400#issuecomment-356390070.

**Comment by jeffswitzer** [comment]
@sis0k0 I modified my webpack.config.ts file to fix the windows paths (see below) but still no luck :S
```
 module: {
            rules: [
                { test: /\.html$|\.xml$/, use: "raw-loader" },

                // tns-core-modules reads the app.css and its imports using css-loader
                { test: /[\/|\\]app\.css$/, use: "css-loader?url=false" },
                { test: /[\/|\\]app\.scss$/, use: ["css-loader?url=false", "sass-loader"] },

                // Angular components reference css files and their imports using raw-loader
                { test: /\.css$/, exclude: /[\/|\\]app\.css$/, use: "raw-loader" },
                { test: /\.scss$/, exclude: /[\/|\\]app\.scss$/, use: ["raw-loader", "resolve-url-loader", "sass-loader"] },

                // Compile TypeScript files with ahead-of-time compiler.
                { test: /.ts$/, use: [
                    "nativescript-dev-webpack/moduleid-compat-loader",
                    { loader: "@ngtools/webpack", options: ngToolsWebpackOptions },
                ]},
            ],
        },
```
Do I need to do something with this const mainSheet = "app.css"; ? I see that it was removed from webpack.typescript.js but maybe it effects how the webpack config gets generated?

**Comment by jeffswitzer** [comment]
Been a while, but I did end up solving this by running ./node_modules/.bin/update-ns-webpack --configs --deps (see [0.9.0 webpack upgrade guide](https://www.nativescript.org/blog/nativescript-webpack-0.9.0-what-changed-and-how-to-upgrade)) after moving my project to a mac. I originally was developing under windows and had issues running that script so was trying alternative methods unsuccessfully to obtain the config files.
